### PR TITLE
added post about modelling generics using dynamic*

### PR DESCRIPTION
**Pull request by gregsdennis**
Inspired by [this conversation](https://github.com/orgs/json-schema-org/discussions/230#discussioncomment-3503774)

Just wanted to get the post written and get thoughts.  I'll come back and add a photo, etc.

### CODE REVIEW

**Review by handrews — COMMENTED** [review]
This is a great example and explanation.  I found one typo :-)

If you would like, I can draw up the dynamic scope behavior of the list or dictionary examples like I did for meta-schemas in the presentation.  The dictionary one would be really interesting to show the two different anchors working at the same time.

It might take until late next week though as I'll be out of town for the long weekend.  But I have the Sketch file for the other diagrams and it's easy enough to modify.

File: pages/posts/dynamicref-and-generics.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  "type": "array",
+  "items": { "$dynamicRef": "#T" }
+}
+```
+
+***NOTE** I used `T` here to match up with the C# `List<T>` to better illustrate what's happening.  You can name it whatever you want.*
+
+If we validate an instance against only this schema, `"$dynamicRef": "#T"` resolves to the subschema with `"$dynamicAnchor": "T"` that we included at `/$defs/content`.  When evaluating just this schema, `$dymamicRef` and `$dynamicAnchor` work just like `$ref` and `$anchor`.
+
+The `"not": true` means that all instances will fail validation.  Typically, to ensure all instances fail validation, we'd use the `false` schema, but in this case, we need to include a dynamic anchor, so a simple `false` doesn't work.  I think `"not": true` is probably the cleanest alternative.  You can also use `"allOf": { false }` if that makes more sense to you.
```
[the remark this answers is not in the document]
Comment (gregsdennis) [reply pages/posts/dynamicref-and-generics.md]: haha.  We're both wrong.  It's actually `"allOf": [ false ]`

**Comment by gregsdennis** [comment]
@handrews I think those examples would be quite useful, yes, but perhaps for UJS or something.  it might be too much for this post.

File: pages/posts/dynamicref-and-generics.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (handrews) [reply pages/posts/dynamicref-and-generics.md]: > haha. We're both wrong. It's actually "allOf": [ false ]

LOL OMG you're right

**Comment by handrews** [comment]
@gregsdennis yeah, good point- don't want to overwhelm with detail when introducing a new topic!

**Review by handrews — APPROVED** [review]

**Review by Relequestual — APPROVED** [review]

File: pages/posts/dynamicref-and-generics.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+The topic in this post can apply to any programming paradigm where you have a defined data model and support for something like generic types.  This will likely be the first in a disjoint series on the relation between data modelling in programming languages and JSON Schema.
+
+Today, I want to cover the idea of generics or templates or some other label you may have heard.  To start, let's cover what I mean by "generic types."  This isn't intended as a lesson on them; I just want to ensure that we're all on the same page.
+
+## Generic Types
+
+By "generic types" I mean the feature in many programming languages to create a type that requires the additional information of one or more secondary types to be complete.  In object-oriented programming, generics can also apply to services as well as data models, but since we use JSON Schema to describe data models, it's we can be fairly certain that the generic types we care about for this use case are wrappers and containers.
```
Comment (Julian) [inline pages/posts/dynamicref-and-generics.md:24]: There's a minor typo here.

> it's we can be fairly certain that the generic types we care about for this use case are wrappers and containers.

I guess the intention is just without the word "it's".

Comment (gregsdennis) [reply pages/posts/dynamicref-and-generics.md:24]: I saw that, meant to change it, then couldn't find it again.  Thanks.

File: pages/posts/dynamicref-and-generics.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+      "$dynamicAnchor": "TValue",
+      "not": true
+    }
+  },
+  "type": "array",
+  "items": {
+    "type": "object",
+    "properties": {
+      "key": { "$dynamicRef": "#TKey" },
+      "key": { "$dynamicRef": "#TValue" }
```
Comment (jdesrosiers) [inline pages/posts/dynamicref-and-generics.md:115]: You have duplicate property names here. I think the second one was intended to be "value".

**Pull request merged**